iceoryx_posh/test/moduletests/test_roudi_portmanager.cpp

```diff
@@ -668,3 +675,57 @@ TEST_F(PortManager_test, PortDestroy)
         }
     }
 }
+
+TEST_F(PortManager_test, UseDestroyInterfaceReturnsNotNullPtrToAcquireInterfacePortData)
+{
+    std::vector<iox::popo::InterfacePortData*> interfaceContainer;
+    std::string itf = "itf";
+
+    // first aquire all possible condition variables
+    for (uint32_t i = 0U; i < iox::MAX_INTERFACE_NUMBER; i++)
+    {
+        auto newItfName = itf + std::to_string(i);
+        auto interp = m_portManager->acquireInterfacePortData(
+            iox::capro::Interfaces::INTERNAL, iox::ProcessName_t(iox::cxx::TruncateToCapacity, newItfName));
+        ASSERT_NE(interp, nullptr);
+        interfaceContainer.push_back(interp);
+    }
+
+    // so now no one should be available
+    {
+        auto interp = m_portManager->acquireInterfacePortData(iox::capro::Interfaces::INTERNAL, m_Process);
+        EXPECT_EQ(interp, nullptr);
+    }
+
+    // set the destroy flag and let the discovery loop take care
+    for (uint32_t i = 0U; i < iox::MAX_INTERFACE_NUMBER; i++)
```

Comment:
@elBoberido  do you want to use this template in all function using container and flags?
i mean ConditionVariablesDestroy, DestroyNodeDataAndAddNewNodeDataSucceeds tests  also.
template <typename vector >
void setDestroyFlagAndClearContainer(vector& Container, int id, int iterations)
{
    for (int i = 0; i < iterations; ++i)
    {
        Container[i]->m_toBeDestroyed.store(true, std::memory_order_relaxed);
    }
    m_portManager->doDiscovery();
    Container.clear();
}
for range based for loop to simplify the code i didn´t understand what you mean exactly.

Reply:
@manel94 yes everywhere where this code snippet is used. You can use the ranged base for loop to get rid of the `iterations` e.g. `for(auto item: container) {...}`. `id` should also not be necessary and `m_portManager->doDiscovery();` should not be part of the method.

Reply:
okay @elBoberido , it will be added in the next update, could you please add remarks need to be refactoring  to all the test file port manager, since i will modify some other things not only my added tests. of course if it´s necessary :D

Reply:
@manel94 it's not right clear to me what the question was 😅

Reply:
Sorry for that @elBoberido,  i mean in this test file i think the old test cases should be refactoring for example test names, using template..., if you could please not just review my added changes but also the old one that help me to refactor all the test file. if it is possible of course.

Reply:
It's mainly regarding the size of the tests. By skimming over the file I wondered if it's possible to consolidate the `arrangement` of some tests in some template methods. And then there is this big `PortDestroy` test, where I'm also not quite sure how to make that more manageable.